### DISCUSSION TITLE
Offer a "Hanging Indent" toggle

### DISCUSSION THREAD

**Discussion by MrBertie** [svonjoi/obsidian-heading-level-indent#13] [Ideas]
Thank you for this useful plugin, I use it regularly to format talk scripts, it is much appreciated.

Would you consider adding a toggle to turn the _Hanging Indent_ on or off?  By default the plugin indents the text below each heading, however in some situations find it easier to have them aligned.

**Default behaviour (Hanging Indent):**
![hanging](https://github.com/svonjoi/obsidian-heading-level-indent/assets/338920/b6d2af22-88b0-48ff-82ea-eefb95aa084e)

**Optional behaviour (No Indent):**
![no hanging](https://github.com/svonjoi/obsidian-heading-level-indent/assets/338920/3a29f4e3-5808-4c95-9091-c12afcd9af5f)

**Suggested new Settings toggle:**
![setting toggle](https://github.com/svonjoi/obsidian-heading-level-indent/assets/338920/57a8ad67-0587-4382-9210-24fe6ac7aaf4)

I went ahead and made the changes I need for my own use in the main.js directly. 😉

**Comment by svonjoi** [comment]
hi, i think this option is cool. Also if you need switch this sometimes, instead of make a setting, its better make a command so there are a possibility to assign a keybinding to it

But first i have to solve the other stuff like visual glitch and support in edit mode, i think i have to rewrite entire plugin creating a codemirror extension to approach this

**Comment by svonjoi** [comment]
Also there are [css snippet](https://github.com/svonjoi/obsidian-heading-level-indent/issues/3#issuecomment-1762873129) that does the same thing without hanging indent, u can use this meanwhile without any glitches, if you dont need hanging indent
